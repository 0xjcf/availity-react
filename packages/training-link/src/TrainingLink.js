import React from 'react';
import PropTypes from 'prop-types';

const shouldProbablyBeInUIKit = {
  paddingTop: '12px',
};

const linkStyles = { fontWeight: 'bold' };

const TrainingLink = ({ name, link }) => (
  <span style={shouldProbablyBeInUIKit} className="ml-auto">
    Need Help?{' '}
    <a href={link} style={linkStyles} className="link" target="_blank" rel="noopener noreferrer">
      Watch a demo
    </a>{' '}
    for {name}
  </span>
);

TrainingLink.propTypes = {
  link: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default TrainingLink;
