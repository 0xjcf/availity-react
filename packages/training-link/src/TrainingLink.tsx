import React from 'react';
import type { CSSProperties } from 'react';

const shouldProbablyBeInUIKit: CSSProperties = { paddingTop: '12px' };
const linkStyles: CSSProperties = { fontWeight: 'bold' };

export type TrainingLinkProps = {
  /**Link to training video.*/
  link?: string;
  /**Name of your app to make the sentence complete.*/
  name?: string;
};

const TrainingLink = ({ name, link }: TrainingLinkProps): JSX.Element => (
  <>
    {name ? (
      <span style={shouldProbablyBeInUIKit} className="ml-auto">
        Need Help?{' '}
        <a href={link} style={linkStyles} className="link" target="_blank" rel="noopener noreferrer">
          Watch a demo <span className="sr-only">(opens in a new tab)</span>
        </a>{' '}
        for {name}
      </span>
    ) : (
      <span style={shouldProbablyBeInUIKit} className="ml-auto">
        Need Help?{' '}
        <a href={link} style={linkStyles} className="link" target="_blank" rel="noopener noreferrer">
          Learn More
        </a>
      </span>
    )}
  </>
);

export default TrainingLink;
