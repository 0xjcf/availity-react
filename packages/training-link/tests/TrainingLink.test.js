import React from 'react';
import renderer from 'react-test-renderer';
import TrainingLink from '..';

describe('TrainingLink', () => {
  test('should not render with link and name', () => {
    const component = renderer.create(
      <TrainingLink link="http://catvidoes.com" name="Appeals" />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
