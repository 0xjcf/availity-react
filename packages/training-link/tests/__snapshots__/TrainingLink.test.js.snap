// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrainingLink should not render with link and name 1`] = `
<span
  className="ml-auto"
  style={
    Object {
      "paddingTop": "12px",
    }
  }
>
  Need Help?
   
  <a
    href="http://catvidoes.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    Watch a demo
  </a>
   
  for 
  Appeals
</span>
`;
